File: DOTNET/C#/VisualC#/Printing/printerResolution/printerResolution/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Drawing.Printing;

namespace printerResolution
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            this.Load += new EventHandler(Form1_Load);
        }

        void Form1_Load(object sender, EventArgs e)
        {
            PrinterSettings setting = new PrinterSettings();
            PageSettings pageSetting = new PageSettings();
            Margins margin = new Margins(200, 200, 200, 200);
            
            setting.DefaultPageSettings.PrinterResolution.Kind = PrinterResolutionKind.High;
            
            pageSetting.PaperSize = new PaperSize("mypaper", 1200, 800);
            setting.DefaultPageSettings.PaperSize = new PaperSize("mypaper", 1200, 800);
            setting.PrintToFile = true;
            setting.PrintFileName = @"d:\documents and settings\axkhan2\desktop\myfile.tiff";
           
            printDocument1.DefaultPageSettings.PrinterSettings = setting;
            printDocument1.DefaultPageSettings = pageSetting;
            printDocument1.DefaultPageSettings.Margins = margin;
            
            printPreviewDialog1.Document = printDocument1;
          
            printPreviewDialog1.ShowDialog();
            //ShowPrintableArea();
            this.Close();
        }

        private void ShowPrintableArea()
        {

            float height = printDocument1.DefaultPageSettings.PrintableArea.Height;
            float width = printDocument1.DefaultPageSettings.PrintableArea.Width;
            float showX = printDocument1.DefaultPageSettings.PrintableArea.X;
            float showY = printDocument1.DefaultPageSettings.PrintableArea.Y;
            MessageBox.Show("Height " + height + " width " + width + " showX " + showX + " show Y " + showY);
            int rX = printDocument1.DefaultPageSettings.PrinterResolution.X;
            int rY = printDocument1.DefaultPageSettings.PrinterResolution.Y;
            string resolutionKind = printDocument1.DefaultPageSettings.PrinterResolution.Kind.ToString();
            MessageBox.Show(" X " + rX + " Y " + rY + " resolution kind " + resolutionKind  );

            string resolution = String.Empty;
            for (int i = 0; i < printDocument1.PrinterSettings.PrinterResolutions.Count - 1; i++)
            {
                resolution += printDocument1.PrinterSettings.PrinterResolutions[i].ToString() + " \n";
            }
            MessageBox.Show(resolution);
            
        }
        public void useparameter()
        {
            string fname, lname, mname;
            fname = "arif";
            lname = "khan";
            mname = "hasna";
        }

        private void printDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e)
        {
            float graphicWidth  = e.PageSettings.PrintableArea.Width;
            float graphicHeight = e.PageSettings.PrintableArea.Height;
            float graphicTop = e.PageSettings.PrintableArea.Top;
            float graphicBottom = e.PageSettings.PrintableArea.Bottom;
            float x = e.PageSettings.PrintableArea.X;
            float y = e.PageSettings.PrintableArea.Y;
            
            MessageBox.Show(" Width " + graphicWidth + " height " + graphicHeight + " top " + graphicTop + " bottom " + graphicBottom + " x " + x + " Y " + y );

            string str = "helloooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo";
            str += "oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo";
            Font fnt = new Font("Arial", 10, FontStyle.Regular); 
           
            e.Graphics.DrawString(str, new Font("Arial", 10, FontStyle.Regular), Brushes.Black, new RectangleF(100, 100, 100, 100), new StringFormat(StringFormatFlags.FitBlackBox));
            //e.Graphics.DrawRectangle(new Pen(Brushes.Black), new Rectangle((int)e.PageSettings.PrintableArea.X,(int) e.PageSettings.PrintableArea.Y, (int)e.PageSettings.PrintableArea.Width,(int) e.PageSettings.PrintableArea.Height));
            //e.Graphics.DrawRectangle(new Pen(Brushes.Blue), new Rectangle((int)e.PageSettings.PrintableArea.Left - e.PageSettings.Margins.Left, (int)e.PageSettings.PrintableArea.Bottom - e.PageSettings.Margins.Bottom, (int)e.PageSettings.PrintableArea.Top - e.PageSettings.Margins.Top,(int)e.PageSettings.PrintableArea.Right - e.PageSettings.Margins.Right));

            e.Graphics.DrawLine(new Pen(Brushes.Black), 100, 100, 300, 300);
            e.Graphics.DrawLine(new Pen(Brushes.Blue), new Point(600, 0), new Point(600, 500));
        }
    }
}

File: DOTNET/C#/VisualC#/PrintingC#/PrintingExample1/PrintingExample1/Form1.vb
Imports System.IO
Imports System.Drawing.Printing
Imports System.Drawing


Public Class Form1
    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

    End Sub

    Private Sub PrintDocument1_PrintPage(ByVal sender As System.Object, ByVal e As System.Drawing.Printing.PrintPageEventArgs) Handles PrintDocument1.PrintPage
        Dim x As Integer = e.MarginBounds.Left
        Dim y As Integer = e.MarginBounds.Top
        
        Dim brush As Brush = New SolidBrush(Color.Red)

        'e.Graphics.DrawLine(Pens.Red, 1, 1, 2000, 2000)
        'e.Graphics.DrawRectangle(Pens.Red, New Rectangle(100, 100, 100, 100))

        Dim fnt As Font = New Font("Arial", 10, FontStyle.Bold)
        'For i As Integer = 20 To 820 Step 20
        '    e.Graphics.DrawString("x 0 y " & i.ToString(), fnt, Brushes.Red, New PointF(0, i))
        '    ' e.Graphics.DrawString("x " & i.ToString() & " y 0", fnt, Brushes.Red, New PointF(i + 10, 20))

        'Next i
        e.Graphics.DrawString("Resume", fnt, Brushes.Blue, New PointF(420, 10))

        e.Graphics.DrawString("Address: Sadabahar, Gilbert Hill Rd,", fnt, Brushes.Blue, New PointF(840, 20))
        e.Graphics.DrawString(Space(17) & "Farooqia Masjid,", fnt, Brushes.Blue, New PointF(840, 40))
        e.Graphics.DrawString(Space(17) & "Anderi(west),", fnt, Brushes.Blue, New PointF(840, 60))
        e.Graphics.DrawString(Space(17) & "Mumbai 40058", fnt, Brushes.Blue, New PointF(840, 80))
        e.Graphics.DrawString("Mobile NO: 9768836054", fnt, Brushes.Blue, New PointF(840, 100))
        e.Graphics.DrawString("Tel NO: 022-32014285", fnt, Brushes.Blue, New PointF(840, 120))

        'e.Graphics.DrawString("x 600 y 600", fnt, Brushes.Red, New PointF(600, 600))

        'e.Graphics.DrawString("x 500 y 500", fnt, Brushes.Red, New PointF(500, 500))
        'e.Graphics.DrawString("x 400 y 400", fnt, Brushes.Red, New PointF(400, 400))
        'e.Graphics.DrawString("x 300, y 300", fnt, Brushes.Red, New PointF(300, 300))
        'e.Graphics.DrawString("x 200 y 200", fnt, Brushes.Red, New PointF(200, 200))
        'e.Graphics.DrawString("x 100 y 100", fnt, Brushes.Red, New PointF(100, 100))
        'e.Graphics.DrawString("x 0 y 100", fnt, Brushes.Red, New PointF(0, 100))
        'e.Graphics.DrawString("x 0 y 200", fnt, Brushes.Red, New PointF(0, 200))

    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'PrintDocument1.DefaultPageSettings.Landscape = True

        Dim printerSetting As PrinterSettings = New PrinterSettings()
        printerSetting.DefaultPageSettings.Landscape = True
        printerSetting.DefaultPageSettings.Margins = New Margins(300, 300, 300, 300)

        Dim pagesetting As PageSettings = New PageSettings(printerSetting)
        pagesetting.Margins = New Margins(100, 100, 200, 200)
        PrintDocument1.PrinterSettings = printerSetting
        PrintPreviewControl1.Document = PrintDocument1

    End Sub
End Class
